runners.go

```diff
@@ -90,6 +124,117 @@ func (s *RunnersService) ListAllRunners(opt *ListRunnersOptions, options ...Opti
+// https://docs.gitlab.com/ce/api/runners.html#get-runner-39-s-details
+func (s *RunnersService) GetRunnerDetails(rid interface{}, options ...OptionFunc) (*RunnersDetails, *Response, error) {
+	runner, err := parseID(rid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("runners/%s", runner)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	var rs *RunnersDetails
+	resp, err := s.client.Do(req, &rs)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return rs, resp, err
+}
+
+// UpdateRunnersDetailsOptions represents the available UpdateRunnersDetails() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/runners.html#update-runner-39-s-details
+type UpdateRunnersDetailsOptions struct {
+	ID          int      `url:"id" json:"id"`
+	Description string   `url:"description,omitempty" json:"description"`
+	Active      bool     `url:"active,omitempty" json:"active"`
+	TagList     []string `url:"tag_list,omitempty" json:"tag_list"`
+	RunUntagged bool     `url:"run_untagged,omitempty" json:"run_untagged"`
+	Locked      bool     `url:"locked,omitempty" json:"locked"`
+	AccessLevel string   `url:"access_level,omitempty" json:"access_level"`
+}
+
+// UpdateRunnersDetails updates runners details
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/runners.html#update-runner-39-s-details
+func (s *RunnersService) UpdateRunnersDetails(rid interface{}, opt *UpdateRunnersDetailsOptions, options ...OptionFunc) (*RunnersDetails, *Response, error) {
+	runner, err := parseID(rid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("runners/%s", runner)
+
+	req, err := s.client.NewRequest("PUT", u, opt, options)
+	var rs *RunnersDetails
+	resp, err := s.client.Do(req, &rs)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return rs, resp, err
+}
+
+// RemoveARunner removes a runner
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/runners.html#remove-a-runner
+func (s *RunnersService) RemoveARunner(rid interface{}, options ...OptionFunc) (*Response, error) {
+	runner, err := parseID(rid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("runners/%s", runner)
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+// ListRunnersJobsOptions represents the available ListRunnersJobs()
+// options. (one of running, success, failed, canceled)
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/runners.html#list-runner-39-s-jobs
+type ListRunnersJobsOptions struct {
+	ListOptions
+	Status *string `url:"status,omitempty" json:"status,omitempty"`
+}
+
+// ListRunnersJobs gets a list of jobs that are being processed or were processed by specified Runner.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/runners.html#list-runner-39-s-jobs
+func (s *RunnersService) ListRunnersJobs(rid interface{}, opt *ListRunnersJobsOptions, options ...OptionFunc) ([]*Job, *Response, error) {
```

Comment:
Let's call this `ListJobRunners` instead... Just like we also have `ListProjectRunners`

Reply:
`ListProjectRunners` lists runners for a project
`ListRunnersJobs` lists jobs for a runner

Don you think that name `ListJobRunners` implies listing runners for a job?

Reply:
Yeah, I can see what you mean... Bu then instead make it `ListRunnerJobs` as it shows the jobs of 1 specific runner. Makes sense?

Reply:
cool ;)